framework/Samplers/Importance.py

```diff
@@ -0,0 +1,110 @@
+
+#Internal Modules------------------------------------------------------------------------------------
+from .ForwardSampler import ForwardSampler
+from utils import utils,randomUtils,InputData, InputTypes
+#Internal Modules End--------------------------------------------------------------------------------
+
+class Importance(ForwardSampler):
+  """
+    MONTE CARLO Sampler
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(Importance, cls).getInputSpecification()
+    samplerInitInput = InputData.parameterInputFactory("samplerInit")
+    limit = InputData.parameterInputFactory("limit", contentType=InputTypes.IntegerType)
+    samplerInitInput.addSub(limit)
+    targetDistribution = InputData.parameterInputFactory("distribution")
+    samplerInitInput.addSub(targetDistribution)
+    samplingTypeInput = InputData.parameterInputFactory("samplingType", contentType=InputTypes.StringType)
+    samplerInitInput.addSub(samplingTypeInput)
+    inputSpecification.addSub(samplerInitInput)
+
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Default Constructor that will initialize member variables with reasonable
+      defaults or empty lists/dictionaries where applicable.
+      @ In, None
+      @ Out, None
+    """
+    ForwardSampler.__init__(self)
+    self.printTag = 'SAMPLER UMBRELLA'
+    self.samplingType = None
+    self.limit = None
+
+  def localInputAndChecks(self,xmlNode, paramInput):
+    """
+      Class specific xml inputs will be read here and checked for validity.
+      @ In, xmlNode, xml.etree.ElementTree.Element, The xml element node that will be checked against the available options specific to this Sampler.
+      @ In, paramInput, InputData.ParameterInput, the parsed parameters
+      @ Out, None
+    """
+    ForwardSampler.readSamplerInit(self,xmlNode)
+    if paramInput.findFirst('samplerInit') != None:
+      if self.limit is None:
+        self.raiseAnError(IOError,self,'Importance sampler '+self.name+' needs the limit block (number of samples) in the samplerInit block')
+    else:
+      self.raiseAnError(IOError,self,'Importance sampler '+self.name+' needs the samplerInit block')
+
+  def localGenerateInput(self, model, myInput):
+    """
+      Provides the next sample to take.
+      After this method is called, the self.inputInfo should be ready to be sent
+      to the model
+      @ In, model, model instance, an instance of a model
+      @ In, myInput, list, a list of the original needed inputs for the model (e.g. list of files, etc.)
+      @ Out, None
+    """
+
+    uSample = self.distDict['uniform'].rvs()
```

Comment:
Is this a hard-coded key? I think usually the distDict is formed as `{variableName: distInstance}`; is `uniform` a variable name here?